python/docs/source/getting_started/installation.rst

```diff
@@ -0,0 +1,119 @@
+standalone locally or as a client to connect to a cluster instead of setting a cluster up.  
+ 
+This page includes the instructions for installing PySpark by using pip, Conda, downloading manually, and building it from the source.
+
+Python Version Supported
+------------------------
+
+Python 3.6 and above.
+
+Using PyPI
+----------
+
+PySpark installation using `PyPI <https://pypi.org/project/pyspark/>`_
+
+.. code-block:: bash
+
+    pip install pyspark
+	
+Using Conda  
+-----------
+
+Conda is an open-source package management and environment management system which is a part of `Anaconda <https://docs.continuum.io/anaconda/>`_ distribution. It is both cross-platform and language agnostic.
+  
+Conda can be used to create a virtual environment from terminal as shown below:
+
+.. code-block:: bash
+
+	conda create -n pyspark_env 
+
+After the virtual environment is created, it should be visible under the list of conda environments which can be seen using the following command:
+
+.. code-block:: bash
+
+	conda env list
+
+The newly created environment can be accessed using the following command:
+
+.. code-block:: bash
+
+	conda activate pyspark_env
+
+In lower Conda version, the following command might be used:
+
+.. code-block:: bash
+
+	source activate pyspark_env
+
+PySpark installation using ``pip`` under Conda environment is official. 
+
+PySpark can be installed in this newly created environment using PyPI as shown before:
+
+.. code-block:: bash
+
+	pip install pyspark
+
+`PySpark at Conda <https://anaconda.org/conda-forge/pyspark>`_ is not the official release.
+
+Official Release Channel
+------------------------
+
+Different flavor of PySpark is available in `the official release channel <https://spark.apache.org/downloads.html>`_.
+Any suitable version can be downloaded and extracted as below:
+
+.. code-block:: bash
+
+    tar xzvf spark-3.0.0-bin-hadoop2.7.tgz
+
+An important step is to ensure ``SPARK_HOME`` environment variable points to the directory where the code has been extracted. 
+The next step is to properly define ``PYTHONPATH`` such that it can find the PySpark and 
+Py4J under ``$SPARK_HOME/python/lib``, one example of doing this is shown below:
+
+.. code-block:: bash
+
+    cd spark-3.0.0-bin-hadoop2.7
+    export SPARK_HOME=`pwd`
+    export PYTHONPATH=$(ZIPS=("$SPARK_HOME"/python/lib/*.zip); IFS=:; echo "${ZIPS[*]}"):$PYTHONPATH
+
+Installing from Source
+----------------------
+
+To install PySpark from source, refer `Building Spark <https://spark.apache.org/docs/latest/building-spark.html>`_.
+
+Steps for defining ``PYTHONPATH`` is same as described in `Official Release Channel <#official-release-channel>`_. 
+
+Dependencies
+------------
+============= ========================= ====================
+Package       Minimum supported version Note
+============= ========================= ====================
+`pandas`      0.23.2                    Optional
+`NumPy`       1.7                       Optional
```

Comment:
and this is required to ML component (but optional in PySpark if you don't use ML).